### DISCUSSION TITLE
Not getting Watermarked on Images

### DISCUSSION THREAD

**Discussion by shub823** [ImageMagick/ImageMagick#3757] [Help]
Hi,
 I am trying to watermark images using the following command:- magick "/repository/thumb/vol0/TESTIMG (3)-S-T_b977eae5b3dafca28abb70d87ca7f753e29eeaa0.JPG"  -set option:wd "%[fx:0.6*w]" \( -background transparent -size "%[wd]x" -stroke "#000b" -fill "#000a" label:"test" -blur 3x3 -stroke none  -fill "#ffff" label:"test" -flatten -rotate "315" \) -gravity center -compose dissolve -define compose:args=50 -composite -quality 50 -geometry +0+25 -compose over -composite "/repository/thumb/vol0/TESTIMG (3)-S-T_b977eae5b3dafca28abb70d87ca7f753e29eeaa0.JPG"

But this command is giving me errors:-   image sequence is required `-composite' @ error/operation.c/CLIListOperationImages/3944.

Version of my Imagemagick:-  7.0.8-3 Q16*86_64 2018

**Comment by dlemstra** [comment]
You are doing both a `-composite` two times. This requires at least two images. You are loading two images but when you do the first `-composite` you will only have a single image after that. And then you cannot do another `-composite`.

**Reply by shub823** [reply]
> You are doing both a `-composite` two times. This requires at least two images. You are loading two images but when you do the first `-composite` you will only have a single image after that. And then you cannot do another `-composite`.

This same code is working in version : 7.0.8-32 Q16*64 2019. But not working in above mentioned version(7.0.8-3 Q16*86_64 2018). Please suggest me on  @dlemstra

**Reply by snibgo** [reply]
Older versions of IM would silently ignore the second (incorrect) "-composite".

So if that did what you want, try removing "-compose over -composite". Does that do what you want?

**Comment by fmw42** [comment]
Another issue you have is that you are using Unix paths with forward slash, but are not escaping your parentheses with backward slash as is required in Unix Imagemagick syntax.
